src/App.tsx

```diff
@@ -1,24 +1,195 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
+import * as postService from './api/todos';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { FilterType } from './types/FilterType';
+import { Error } from './types/Error';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoError } from './components/TodoError';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+const USER_ID = 11260;
+
+const getVisibleTodos = (todos: Todo[], status: FilterType) => {
+  return todos
+    .filter(todo => {
+      switch (status) {
+        case FilterType.Completed:
+          return todo.completed;
+
+        case FilterType.Active:
+          return !todo.completed;
+
+        default:
+          return true;
+      }
+    });
```

Comment:
```suggestion
      switch (status) {
        case FilterType.Completed:
          return todos.filter(todo => todo.completed);

        case FilterType.Active:
          return todos.filter(todo => !todo.completed);

        default:
          return todos;
      }
```
consider to filter todos inside switch. it wil avoid redundant iteration in case if no need to filter todos.